### ISSUE TITLE
Subscription filter

### ISSUE THREAD

**Issue by innomal** [doug-martin/nestjs-query#317]
I have been trying to apply filters to subscription but as soon as any sort of filter as applied subscription doesn't receives any update, I have attached two corresponding mutation and subscription i use to check filter:
```
mutation {
  updateOneUser(input:{id:1, update:{ lastName:"ss" } }) {
    firstName
    lastName
  }
}
```
```
subscription {
  updatedOneUser(input: { filter: { id: { eq:"1" } } }) {
    id
    firstName
    lastName
  }
}
```
With filter id filter being removed from subscription it receives updates. But on all records.

Thanks!

**Comment by doug-martin** [comment]
What if you change to `id: { eq: 1 }` ?  The eq comparison will be strict eq, I think in this case you are using a string for the ID but it is actually a number.

**Comment by innomal** [comment]
Thanks for response,
I tried that too when i was testing but had no luck. I also tried with other filters (in, gt etc).
Also tried with filtering other fields but couldn't get any response as soon as any sort of filter was applied.

**Comment by doug-martin** [comment]
@innomal I'm curious is your `id` field using the `ID` graphql type? 

In local testing it looks like that always serializes to a string, which causes this to break because of the strict equal. Once I changed the `id` type to `Int` the subscription filtered properly.

**Comment by innomal** [comment]
Yes it is set to `ID`, I have attached dto class for reference.
```
@ObjectType('User')
export class UserDTO {
    
    @FilterableField(() => ID)
    id!: number;
    
    @FilterableField()
    mobileNumber?:number;

    @Field()
    firstName?: string;
    
    @FilterableField()
    lastName?: string;
}
```
I can confirm the issue i am experiencing seems to be about `ID` field. I could get a subscription from a string field filter correctly but not yet with id.
Also just an idea but maybe it could be useful to be possible to filter based on fields. Like only if `lastName` & `firstName` was changed emit an update.
